src/overlays/actors/ovl_En_Ru/z_en_ru.c

```diff
@@ -76,34 +86,285 @@ static DamageTable D_80A39488 = {
+    { 0xFA0, 4, 1, 3 },
+    { 0x1770, 4, 1, 6 },
+};
+
+static Vec3f sHeadFocusPos = {
+    800.0f,
+    0,
+    0,
+};
+
+static Vec3f sBodyFocusPos = {
+    0,
+    0,
+    0,
+};
+
+// copy of displaylist found in En_Zo
+static Gfx sTransparencyDlist[] = {
+    gsDPSetRenderMode(AA_EN | Z_CMP | Z_UPD | IM_RD | CLR_ON_CVG | CVG_DST_WRAP | ZMODE_XLU | FORCE_BL |
+                          G_RM_FOG_SHADE_A,
+                      AA_EN | Z_CMP | Z_UPD | IM_RD | CLR_ON_CVG | CVG_DST_WRAP | ZMODE_XLU | FORCE_BL |
+                          GBL_c2(G_BL_CLR_IN, G_BL_A_IN, G_BL_CLR_MEM, G_BL_1MA)),
+    gsDPSetAlphaCompare(G_AC_THRESHOLD),
+    gsSPEndDisplayList(),
+};
+
+s32 EnRu_ChangeAnimation(SkelAnime* skelAnime, s16 animIndex) {
+    s16 lastFrame;
+    s32 ret = false;
+
+    if ((animIndex >= 0) && (animIndex < ARRAY_COUNT(sAnimations))) {
+        lastFrame = sAnimations[animIndex].frameCount;
+
+        ret = true;
+        if (lastFrame < 0) {
+            lastFrame = Animation_GetLastFrame(sAnimations[animIndex].animation);
+        }
+
+        Animation_Change(skelAnime, sAnimations[animIndex].animation, sAnimations[animIndex].playSpeed,
+                         sAnimations[animIndex].startFrame, lastFrame, sAnimations[animIndex].mode,
+                         sAnimations[animIndex].morphFrames);
+    }
+
+    return ret;
+}
+
+// En_Zo has a copy of this function
+s32 EnRu_PlayWalkingSound(EnRu* this, PlayState* play) {
+    u8 leftWasGrounded;
+    u8 rightWasGrounded;
+    s32 waterSfxId;
+    s16 sfxId;
+    u8 isFootGrounded;
+
+    leftWasGrounded = this->isLeftFootGrounded;
+    rightWasGrounded = this->isRightFootGrounded;
+
+    if (this->actor.bgCheckFlags & 0x20) {
+        if (this->actor.depthInWater < 20.0f) {
+            waterSfxId = NA_SE_PL_WALK_WATER0 - SFX_FLAG;
+        } else {
+            waterSfxId = NA_SE_PL_WALK_WATER1 - SFX_FLAG;
+        }
+
+        sfxId = waterSfxId + SFX_FLAG;
+
+    } else {
+        sfxId = SurfaceType_GetSfx(&play->colCtx, this->actor.floorPoly, this->actor.floorBgId) + SFX_FLAG;
+    }
+
+    this->isLeftFootGrounded = isFootGrounded = SubS_IsFloorAbove(play, &this->leftFootPos, -6.0f);
+
+    if ((this->isLeftFootGrounded) && (!leftWasGrounded) && (isFootGrounded)) {
+        Actor_PlaySfxAtPos(&this->actor, sfxId);
+    }
+
+    this->isRightFootGrounded = isFootGrounded = SubS_IsFloorAbove(play, &this->rightFootPos, -6.0f);
+    if ((this->isRightFootGrounded) && (!rightWasGrounded) && (isFootGrounded)) {
+        Actor_PlaySfxAtPos(&this->actor, sfxId);
+    }
+
+    return false;
+}
+
+// this func is only ever called with eyeStateMax == 3
+void EnRu_UpdateEyes(EnRu* this, s32 eyeStateMax) {
+    if (DECR(this->blinkTimer) == 0) {
+        this->eyeState++;
+        if (this->eyeState >= eyeStateMax) {
+            this->eyeState = 0;
+            this->blinkTimer = Rand_S16Offset(30, 30);
+        }
+    }
+}
+
+void EnRu_CollisionCheck(EnRu* this, PlayState* play) {
```

Comment:
We usually call this sort of function `UpdateCollider`(`s`) or `UpdateCollision`